### PR TITLE
Burials | Date alert message isn't announced to screen reader users

### PR DESCRIPTION
**Pull request by jerekshoe**
## Original Issue(s)
https://github.com/department-of-veterans-affairs/va.gov-team/issues/42249

[URL of fix](/burials-and-memorials/application/530/veteran-information/burial)

## Expected behavior
When filling out the Veterans Burial Date, if the user enters a date that is more than 2 years ago, a `va-alert` will appear below the date fields informing them that the burial date must be within 2 years of todays date in order for them to be eligible to receive a non-service-connected allowance. The alert content should be read aloud by a screenreader

## Current behavior
Currently when the user enters a date that is more than 2 years ago, the alert appears, but the screenreader ignores it

## Your fix
I updated the code to use `va-alert` instead of the `usa-alert` CSS class and added `aria-live="polite"` to it. I also had to delay the rendering of the alert by 1000ms because VoiceOver (on macOS at least) reads the last number entered into the year input, followed by the whole number. Without the delay, VoiceOver will read the last number, then start to read the alert content, then immediately read the whole year input.

I also made some updates to the styling as part of the tickets request as well

## How has this been tested?
* Navigate to `/burials-and-memorials/application/530/veteran-information/burial`
* Activate VoiceOver
* Enter a date more than 2 years prior to todays date in the Date of Burial field
* VoiceOver should read the last number entered, then the whole year, then the alert content

### Current behavior
current

### Expected behavior
delayed

## Acceptance criteria
- [x] The dynamic alert is announced using `aria-live="polite"`
- [x] An icon with alt text or a text label is provided to represent the urgency the background color yellow communicates (e.g. "**Notice**: If filing for a non-service-connected allowance, the Veteran’s burial date must be no more than 2 years from the current date. Find out if you still qualify..."
- [x] The link text should include "(opens in new tab)" to give magnification users a heads up that it will be opening in a new tab

## Definition of done
- [ ] *

### CODE REVIEW

File: src/applications/burials/helpers.jsx [hunk 1]
```diff
@@ -163,18 +163,32 @@ export const transportationWarning = (
...
-  </div>
-);
+export const BurialDateWarning = () => {
+  const [shouldRender, setShouldRender] = useState(false);
+
+  useEffect(() => {
+    // NOTE: If we don't wait at least 900ms to render,
+    // the alert content gets overridden by the year content.
+    // Using 1000ms to give a bit of padding
+    const timeout = setTimeout(() => setShouldRender(true), 1000);
```
Comment (Mottie) [inline src/applications/burials/helpers.jsx:173]: I wonder if 1000ms is enough for a phone with a slow processor? Not sure how to test for this...

File: src/applications/burials/helpers.jsx [hunk 2]
```diff
@@ -163,18 +163,32 @@ export const transportationWarning = (
...
+    return () => clearTimeout(timeout);
+  }, []);
+
+  return (
+    shouldRender && (
+      <va-alert aria-live="polite" background-only show-icon status="warning">
```
Comment (Mottie) [inline src/applications/burials/helpers.jsx:179]: @joshkimux Correct me if I'm wrong, please, but doesn't an `aria-live` element need to exist on the page before adding & removing content?

**Review by Mottie — APPROVED** [review]
Approving, since I had questions instead of actual requests for changes.

File: src/applications/burials/helpers.jsx (hunk 2 above)
Comment (jerekshoe) [reply src/applications/burials/helpers.jsx:179]: Does it always have to exist on the page or is it enough for it to exist right before the element is added. I’m not 100% sure how to have the form system hide only part of the content of the ui:description

Comment (joshkimux) [reply src/applications/burials/helpers.jsx:179]: Ideally we want it to exist on the page; it performs best that way. If we can't do that, setting a delay usually between the container `div` being added and the content being added will do the trick.

**Review by Mottie — APPROVED** [review]

**Review by joshkimux — COMMENTED** [review]
Just one little bit to address related to alt text for the colored background! Other than that, LGTM 🚀

**Pull request merged**